source/SpinalHDL/Data types/bool.rst

```diff
@@ -9,7 +9,52 @@ Bool
 Description
 ^^^^^^^^^^^
 
-The ``Bool`` type corresponds to a boolean value (True or False).
+The ``Bool`` type corresponds to a boolean value (True or False) or a single bit/wire
+used in a hardware design.  While named similarly it should not be confused with
+Scala `Boolean` type which does not describe hardware but truth values in the Scala
+generator code.
+
+An important concept and rule-of-thumb to understand is that the Scala
+`Boolean` type is used in places where ahead-of-time HDL code-generation
+decision making is occuring in Scala code.  Like any regular program it affects
+execution of the Scala program that is SpinalHDL at the time the program is
+being run to perform HDL code generation.
+
+Therefore the value of a Scala `Boolean` can not be observed from hardware,
+because it only exists ahead-of-time in the SpinalHDL program at the time of
+HDL code-gen.
+
+In scenarios where you might need this for your design, for example to pass a
```

Comment:
The general difference between scala types and SpinalHDL types would make sense in the introduction on datatypes.

Reply:
Maybe it would,  I shall re-read the datatypes sections again, but not from a typo/grammar improvement stance, more of a novice/beginner understanding it stance.

I certainly did not find the Scala datatype/class diagram and other info useful until I first needed to use Scala generics on Data `[T <: Data]` in some code.  Only then did I found it super useful.  I did not in anyway find it useful to translate existing Verilog/HDL knowledge into SpinalHDL knowledge concerning data.

There are probably only around 5 major areas that are worth an explanation like this before there should be enough knowledge to understand reference material.  Bool is defiantly one place.  Bits/Ints is another (which is conceptually a `bus`),  Mux/when/conditions is another, Register is another, combinational expressions and signal direction another.

Reply:
I actually found it helpful in the beginning, it's very similar to diagrams that I was used to from VHDL...

Let's keep it this for now and I'll move it when I fix the order of the chapters.